src/main/java/com/iexec/worker/executor/TaskManagerService.java

```diff
@@ -59,33 +63,39 @@ public class TaskManagerService {
     private final ContributionService contributionService;
     private final RevealService revealService;
     private final ComputeManagerService computeManagerService;
-    private final TeeSconeService teeSconeService;
+    private final TeeServicesManager teeServicesManager;
     private final DataService dataService;
     private final ResultService resultService;
     private final DockerService dockerService;
     private final SubscriptionService subscriptionService;
+    private final SmsClientProvider smsClientProvider;
+    private final TeeWorkflowConfigurationService teeWorkflowConfigurationService;
 
     public TaskManagerService(
             WorkerConfigurationService workerConfigurationService,
             IexecHubService iexecHubService,
             ContributionService contributionService,
             RevealService revealService,
             ComputeManagerService computeManagerService,
-            TeeSconeService teeSconeService,
+            TeeServicesManager teeServicesManager,
             DataService dataService,
             ResultService resultService,
             DockerService dockerService,
-            SubscriptionService subscriptionService) {
+            SubscriptionService subscriptionService,
+            SmsClientProvider smsClientProvider,
+            TeeWorkflowConfigurationService teeWorkflowConfigurationService) {
```

Comment:
Constructor is heavy. How about moving some newly introduced logic to `SmsService` and/or `TeeServicesManager`?

Reply:
It looks better into `TeeService` (which can me changed from an interface to an abstract class). Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/7dd0d140d9802e9dc547b811c3689989c49109f1.

Reply:
Yes, we tried to avoid an abstract `TeeService` until here (interfaces are better), but it looks a good compromise for today, thanks :)

Reply:
By doing this you can migrate `SgxService` field from `BlaBlaTeeService`s to parent class `TeeService`.

Reply:
Sure, please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/c7338798e2cb37d25e45da4ff4d42ecef658b1d6.

Reply:
Nice!
Can you move both
```
    @Override
    public boolean isTeeEnabled() {
        return sgxService.isSgxEnabled();
    }
```
 to `TeeService` while we don't have a third TEE provider?

Reply:
Sure, see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/d4700ede17b31d1e2609a4821bce19ae1d47eabb.

Reply:
Thanks :)